### PR TITLE
#8 - Updates for Julia v1.0

### LINKED ISSUE #8: Updates for v1.0
- some missing imports form the standard library
- fix some deprecated names in the benchmarks

### PR DESCRIPTION
**Pull request by mforets**
Closes #8.

### CODE REVIEW

**Comment by mforets** [comment]
with this PR 

```julia
# file benchmark/benchmarks.jl

include("../src/SpaceStation/iss.jl")  #works (*)
include("../src/Building/build.jl") # breaks since SX needs to be updated
```

in my machine however it breaks with an unrelated message:

```julia
$ julia1 -e 'using PkgBenchmark, Reachability; results = benchmarkpkg("ARCH2018_RE")'
PkgBenchmark: Running benchmarks...
Computing successors 100%|██████████████████████████████| Time: 0:00:03
Computing successors  14%|████▎                         |  ETA: 0:00:56PkgBenchmark: creating benchmark tuning file /Users/forets/.julia/dev/ARCH2018_RE/bench
mark/tune.json...
OpenBLAS : Program will terminate because you tried to start too many threads.
OpenBLAS : Program will terminate because you tried to start too many threads.
ERROR: failed process: Process(`/Applications/Julia-1.0.app/Contents/Resources/julia/bin/julia --project=/Users/forets/.julia/environments/v1.0 --depwarn=no -
-code-coverage=none --color=no --compiled-modules=yes -e 'using PkgBenchmark
PkgBenchmark._runbenchmark_local("/Users/forets/.julia/dev/ARCH2018_RE/benchmark/benchmarks.jl", "/var/folders/b_/qy7kpq613ld0g06jy9g16hz00000gp/T/juliavSHEaX
", "/Users/forets/.julia/dev/ARCH2018_RE/benchmark/tune.json", false )

'`, ProcessSignaled(11)) [0]
Stacktrace:
 [1] uv_error at ./libuv.jl:85 [inlined]
 [2] unlink(::String) at ./file.jl:722
 [3] #rm#9(::Bool, ::Bool, ::Function, ::String) at ./file.jl:253
 [4] #rm at ./none:0 [inlined]
 [5] _withtemp(::getfield(PkgBenchmark, Symbol("##19#21")){Bool,String}, ::String) at /Users/forets/.julia/packages/PkgBenchmark/6RhAC/src/util.jl:6
 [6] (::getfield(PkgBenchmark, Symbol("#do_benchmark#20")){Bool,String,String})() at /Users/forets/.julia/packages/PkgBenchmark/6RhAC/src/runbenchmark.jl:86
 [7] #benchmarkpkg#18(::Nothing, ::Nothing, ::Bool, ::String, ::Function, ::String, ::BenchmarkConfig) at /Users/forets/.julia/packages/PkgBenchmark/6RhAC/src
/runbenchmark.jl:108
 [8] benchmarkpkg at /Users/forets/.julia/packages/PkgBenchmark/6RhAC/src/runbenchmark.jl:41 [inlined] (repeats 2 times)
 [9] top-level scope at none:0
```

(i have tried to increase the number of threads but still. . .)

**Comment by mforets** [comment]
there seems to be an upstream solution from yesterday (issue 30725 in https://github.com/JuliaLang/julia), i would have to check.

**Comment by schillic** [comment]
If this error only occurs because of `PkgBenchmark`, we could add an alternative timing mechanism.

File: src/SpaceStation/iss.jl [hunk 1]
```diff
@@ -5,9 +5,9 @@ SUITE["ISS"] = BenchmarkGroup()
 # ==============================
 # Load model
 # ==============================
-file = @relpath "iss.mat"
-file = matopen(file)
+file = matopen(@relpath "iss.mat")
 A, B, C = sparse(read(file, "A")), read(file, "B"), Matrix(read(file, "C")[3, :]')
+n = size(A, 1)
```
Comment (schillic) [inline src/SpaceStation/iss.jl:10]: Now we can use `n` instead of `size(A, 1)` three lines below.

**Review by schillic — APPROVED** [review]

**Comment by mforets** [comment]
> If this error only occurs because of PkgBenchmark, we could add an alternative timing mechanism.

i found this issue even without benchmarks, for example running the hybrid test suite in Reachability

File: src/SpaceStation/iss.jl [hunk 2]
```diff
@@ -5,9 +5,9 @@ SUITE["ISS"] = BenchmarkGroup()
...
 Cvec = C[:]
 time_horizon = 20.0
 X0 = BallInf(zeros(size(A, 1)), 0.0001)
```
Comment (mforets) [inline src/SpaceStation/iss.jl]:
```suggestion
X0 = BallInf(zeros(n), 0.0001)
```

**Comment by schillic** [comment]
```julia
┌ Warning: Package ARCH2018_RE does not have SparseArrays in its dependencies:
│ - If you have ARCH2018_RE checked out for development and have
│   added SparseArrays as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with ARCH2018_RE
└ Loading SparseArrays into ARCH2018_RE from project dependency, future warnings for ARCH2018_RE are suppressed.
```

**Pull request merged**